### PR TITLE
🚀 Add age of block expression

### PR DESCRIPTION
**Pull request by AyhamAl-Ali**
### Description

This PR adds a new expression `((max[imum]|) age) of %block%` used for ageable blocks such as crops

---
**Target Minecraft Versions:** 1.13+
**Requirements:** Minecraft 1.13+
**Related Issues:** #4534 #3068 (partially, might need to implement all that if i can)

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/ExprBlockAge.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		isMax = parseResult.hasTag("max");
+		setExpr((Expression<Block>) exprs[0]);
+		return true;
+	}
+
+	@Override
+	@Nullable
+	public Number convert(Block block) {
+		BlockData bd = block.getBlockData();
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprBlockAge.java:61]: This method isn't there on 1.9, only make the expression register if it (and the interface Ageable, if needed) exist, and add docs accordingly

File: src/main/java/ch/njol/skript/expressions/ExprBlockAge.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			else
+				return ((Ageable) bd).getAge();
+		}
+		return null;
+	}
+	
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		return mode == ChangeMode.SET || mode == ChangeMode.RESET ? CollectionUtils.array(Number.class) : null;
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprBlockAge.java:74]: Should return null if isMax

File: src/main/java/ch/njol/skript/expressions/ExprBlockAge.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	}
+	
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		if (mode == ChangeMode.SET && (delta == null || delta[0] == null)) {
+			return;
+		}
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprBlockAge.java:81]:
```suggestion
		if (mode == ChangeMode.SET && (delta == null || delta[0] == null))
			return;
```

File: src/main/java/ch/njol/skript/expressions/ExprBlockAge.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.block.Block;
+import org.bukkit.block.data.Ageable;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Block Age")
+@Description({"Returns the age of a block. 'Age' represents the different growth stages that a crop-like block can go through.",
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprBlockAge.java:38]: First sentence should include something about the age or the maximum age

**Review by TPGamesNL — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprBlockAge.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+			else
+				return ((Ageable) bd).getAge();
+		}
+		return null;
+	}
+	
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		return (mode == ChangeMode.SET || mode == ChangeMode.RESET) && !isMax ? CollectionUtils.array(Number.class) : null;
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/expressions/ExprBlockAge.java:78]: This looks slightly better
```suggestion
		return !isMax && (mode == ChangeMode.SET || mode == ChangeMode.RESET) ? CollectionUtils.array(Number.class) : null;
```

**Review by TPGamesNL — APPROVED** [review]

**Review by TheLimeGlass — APPROVED** [review]

**Pull request merged**